azurerm/provider.go

```diff
@@ -129,168 +129,170 @@ func Provider() terraform.ResourceProvider {
+			"azurerm_kubernetes_cluster":                                 resourceArmKubernetesCluster(),
+			"azurerm_lb":                                                 resourceArmLoadBalancer(),
+			"azurerm_lb_backend_address_pool":                            resourceArmLoadBalancerBackendAddressPool(),
+			"azurerm_lb_nat_rule":                                        resourceArmLoadBalancerNatRule(),
+			"azurerm_lb_nat_pool":                                        resourceArmLoadBalancerNatPool(),
+			"azurerm_lb_probe":                                           resourceArmLoadBalancerProbe(),
+			"azurerm_lb_rule":                                            resourceArmLoadBalancerRule(),
+			"azurerm_local_network_gateway":                              resourceArmLocalNetworkGateway(),
+			"azurerm_log_analytics_solution":                             resourceArmLogAnalyticsSolution(),
+			"azurerm_log_analytics_workspace":                            resourceArmLogAnalyticsWorkspace(),
+			"azurerm_logic_app_action_custom":                            resourceArmLogicAppActionCustom(),
+			"azurerm_logic_app_action_http":                              resourceArmLogicAppActionHTTP(),
+			"azurerm_logic_app_trigger_custom":                           resourceArmLogicAppTriggerCustom(),
+			"azurerm_logic_app_trigger_http_request":                     resourceArmLogicAppTriggerHttpRequest(),
+			"azurerm_logic_app_trigger_recurrence":                       resourceArmLogicAppTriggerRecurrence(),
+			"azurerm_logic_app_workflow":                                 resourceArmLogicAppWorkflow(),
+			"azurerm_managed_disk":                                       resourceArmManagedDisk(),
+			"azurerm_management_lock":                                    resourceArmManagementLock(),
+			"azurerm_management_group":                                   resourceArmManagementGroup(),
+			"azurerm_metric_alertrule":                                   resourceArmMetricAlertRule(),
+			"azurerm_monitor_action_group":                               resourceArmMonitorActionGroup(),
+			"azurerm_mysql_configuration":                                resourceArmMySQLConfiguration(),
+			"azurerm_mysql_database":                                     resourceArmMySqlDatabase(),
+			"azurerm_mysql_firewall_rule":                                resourceArmMySqlFirewallRule(),
+			"azurerm_mysql_server":                                       resourceArmMySqlServer(),
+			"azurerm_mysql_virtual_network_rule":                         resourceArmMySqlVirtualNetworkRule(),
+			"azurerm_network_interface":                                  resourceArmNetworkInterface(),
+			"azurerm_network_interface_backend_address_pool_association": resourceArmNetworkInterfaceBackendAddressPoolAssociation(),
+			"azurerm_network_interface_nat_rule_association":             resourceArmNetworkInterfaceNatRuleAssociation(),
```

Comment:
@katbyte given we're doing a resource for application gateway associations too, I think this needs `lb_` between `network_interface` and `backend/nat`- WDYT?